src/test/javascript/spec/component/programming-exercise/programming-exercise-reset-dialog.component.spec.ts

```diff
@@ -0,0 +1,234 @@
+            jest.spyOn(activeModal, 'dismiss').mockImplementation();
+            jest.spyOn(alertService, 'success').mockImplementation();
+
+            comp.handleResetResponse();
+
+            expect(alertService.success).toHaveBeenCalledWith('artemisApp.programmingExercise.reset.successMessage');
+            expect(activeModal.dismiss).toHaveBeenCalledWith(true);
+        });
+
+        it('should not be called when there is an error in the reset response', fakeAsync(() => {
+            const errorResponse = throwError({ status: 500 });
+            jest.spyOn(programmingExerciseService, 'reset').mockReturnValue(errorResponse);
+            jest.spyOn(comp, 'handleResetResponse').mockImplementation();
+
+            comp.resetProgrammingExercise();
+
+            tick();
+
+            expect(comp.handleResetResponse).not.toHaveBeenCalled();
+            expect(comp.resetInProgress).toBeFalse();
+        }));
+    });
+
+    describe('areSecurityChecksFulfilled', () => {
+        it('should return false when confirmation text is empty', () => {
+            comp.confirmText = '';
+
+            expect(comp.areSecurityChecksFulfilled).toBeFalse();
+        });
+
+        it('should return false when confirmation text is not filled correctly', () => {
+            comp.confirmText = 'Incorrect Name';
+
+            expect(comp.areSecurityChecksFulfilled).toBeFalse();
+        });
+
+        it('should return true when confirmation text is filled correctly', () => {
+            comp.confirmText = 'Programming Exercise';
+
+            expect(comp.areSecurityChecksFulfilled).toBeTrue();
+        });
+    });
+
+    describe('hasSelectedOptions', () => {
+        it('should return false when all options are set to false', () => {
+            comp.programmingExerciseResetOptions = {
+                deleteBuildPlans: false,
+                deleteRepositories: false,
+                deleteParticipationsSubmissionsAndResults: false,
+                recreateBuildPlans: false,
+            };
+
+            expect(comp.hasSelectedOptions).toBeFalse();
+        });
+
+        it('should return true when at least one option is set to true', () => {
+            const options = {
+                deleteBuildPlans: true,
+                deleteStudentRepositories: true,
+                deleteStudentParticipationsSubmissionsAndResults: true,
+                recreateBuildPlans: true,
+            };
+
+            for (const option in options) {
+                comp.programmingExerciseResetOptions = {
+                    deleteBuildPlans: false,
+                    deleteRepositories: false,
+                    deleteParticipationsSubmissionsAndResults: false,
+                    recreateBuildPlans: false,
+                };
+                comp.programmingExerciseResetOptions[option] = true;
+                expect(comp.hasSelectedOptions).toBeTrue();
+            }
+        });
```

Comment:
Maybe you can create a parametrized test here instead?

Reply:
TIL, thanks!